example/exampleContour.html

```diff
@@ -145,51 +145,40 @@
           lineJoin: "bevel",
         },
         encoding: {
-          // Keep this here for now to keep line mixin happy
-          color: {},
-
-          // TODO(joe): Implement encoding in this format
-          // color: {
-          //   field: "is_major_contour",
-          //   type: "nominal",
-          //   scale: { 
-          //     domain: [false, true], 
-          //     range: ["#999999", "#999999"] 
-          //   }
-          // },
-          // strokeWidth: { 
-          //   field: "is_major_contour", 
-          //   type: "nominal", 
-          //   scale: { 
-          //     domain: [false, true], 
-          //     range: [1, 2] 
-          //   } 
-          // },
-          // opacity: {
-          //   field: "is_major_contour",
-          //   type: "nominal",
-          //   scale: { 
-          //     domain: [false, true], 
-          //     range: [0.75, 0.8] 
-          //   } 
-          // },
-
-          // Contour stuff
-          stroke: {
-            minor: {
-              width: 1,
-              color: "#666666",
-              opacity: 0.75
-            },
-            major: {
-              width: 2,
-              color: "#666666",
-              opacity: 0.8
+          geocol: "contour_lines",
+          color: {
+            field: "is_major",
```

Comment:
oh, looking further below in the `buildContourSQL` method, it seems you are saying this "field" value will define the name of that column. That certainly does make things less redundant, but what happens if the `field` is misnamed/different in the `strokeWidth`/`opacity`?

Anyway, I like this field approach to reduce redundancy, but we would need a plan to coordinate with the other encodings that might want to make use of the same `is_major` field.

Reply:
Yeah wasn't quite sure how to handle that, maybe just move it to a root property of encoding, like `encoding.majorMinorField`? Or are there any existing examples of something like this?

Reply:
Oh you mentioned putting it in the data block, i'll see what this looks like!

Reply:
Hm so I think it still makes sense to set it up like this:
```
const majorMinorField = "is_major"
...
contourLayer.setState({
        data: [{
          type: "contour",
          majorMinorField,
          ...
        }],
        encoding: {
          color: {
             field: majorMinorField,
             ...
          },
          ...
        }
    })
```

This leaves it open in the future if we wanted to setup other things to drive color/strokeWidth/opacity, but charting will only get the field name from one place. Lemme know if that makes sense to you

Reply:
yes, I was going to suggest something similar. It might also make sense to namespace the major/minor-related properties in a sub-block, so:

```
{
  "type": "contour",
  ...,
  "intervals": {
    "major": <major interval>,
    "minor": <minor interval>,
    "field_name": "is_major"
  },
  ...
}
```

Reply:
maybe `intervalTypeFieldName`? Can't go wrong with super descriptive?

Reply:
Updated the format to match above with the descriptive field name property, if you prefer `field` i'm happy to change it back

Reply:
yeah, it helps to be more descriptive with the naming, even if it is more verbose.